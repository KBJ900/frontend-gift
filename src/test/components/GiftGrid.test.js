import React from "react";
import { shallow } from "enzyme";
import { GiftGrid } from "../../components/GiftGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el <GiftGrid/>", () => {
  const category = "One Punch";

  test("Debe de mostrarse correectamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GiftGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostar items cuando se cargan imagenes useFetchGifts", () => {
    const gifs = [
      {
        id: "ABC",
        url: "http://localhost/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "123",
        url: "http://localhost/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GiftGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
