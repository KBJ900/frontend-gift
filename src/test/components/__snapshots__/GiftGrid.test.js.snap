// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GiftGrid/> Debe de mostar items cuando se cargan imagenes useFetchGifts 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="http://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GiftGrid/> Debe de mostrarse correectamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__fadeIn"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
